test/framework/src/main/java/org/elasticsearch/test/rest/yaml/section/DoSection.java

```diff
@@ -307,16 +310,20 @@ void checkWarningHeaders(final List<String> warningHeaders, final Version master
                      * We skip warnings related to types deprecation and transform rename so that we can continue to run the many
                      * mixed-version tests that used typed APIs.
                      */
-                } else if (expected.remove(message) == false) {
+                } else if (expected.contains(message) == false) {
```

Comment:
@nik9000 Since you wrote the original logic, thought I would ping you.

This allows for the same warning to be mentioned multiple times in the headers. This is required for ML since there are times when a warning appears more than once. 

The number of times a warning appears is non-deterministic in multi-node tests. 

This is because for ML we occasionally need to route to specific nodes (e.g. to a master node) and warnings can appear on parsing XContent and on serializing between nodes.

Reply:
Also, this PR is against 7.x. If approved and merged, I am going to forward-port this change.

Reply:
@nik9000 the issue is because the warning is added via different nodes. One is when the object is deserialized via XContent. The other is when it is streamed across to another node:

```
WARNING: request [PUT http://[::1]:59141/_ml/datafeeds/mixed-cluster-datafeed-with-aggs?error_trace=true] returned 2 warnings:
[299 Elasticsearch-7.4.0-22e1767283e61a198cb4db791ea66e3f11ab9910 "[interval] on [date_histogram] is deprecated, use [fixed_interval] or [calendar_interval] in the future."],
[299 Elasticsearch-7.7.0-SNAPSHOT-ea4c2ef5cb553bcf5fff8226c0878411fa9b6a2e "[interval] on [date_histogram] is deprecated, use [fixed_interval] or [calendar_interval] in the future."]
```

I am not sure how to prevent this as they are different thread contexts altogether.

Reply:
:+1: I filed #53123. I'll see about making an "allowed warnings".